index.js

```diff
@@ -1,35 +1,46 @@
 const express = require("express");
 const app = express();
 const mongoose = require("mongoose");
+const JwtService = require("./services/JwtService");
+const UserService = require("./services/UserService");
+const Error = require("./models/Error");
+require("dotenv").config;
 
 app.use(express.urlencoded({ extended: true }));
 app.use(express.json());
 
 mongoose
-    .connect(process.env.MONGO, {
-        useNewUrlParser: true,
-        useUnifiedTopology: true,
-        useCreateIndex: true,
-        useFindAndModify: false,
-    })
-    .then(() => console.log("Connected to mongodb"))
-    .catch((err) => console.log(err));
+  .connect(process.env.MONGO, {
+    useNewUrlParser: true,
+    useUnifiedTopology: true,
+    useCreateIndex: true,
+    useFindAndModify: false,
+  })
+  .then(() => console.log("Connected to mongodb"))
+  .catch((err) => console.log(err));
 
 const port = 3000;
 
-app.get("/", async (req, res) => {
-    //! Only for testing
+// Adds the loggedIn user to the response's local vars for easy access to
+// other routes.
+app.use(async (req, res, next) => {
+  if (req.headers["authorization"]) {
+    const accessToken = req.headers["authorization"];
     try {
-        const users = await require("./services/UserService").getAllUsers();
-        res.status(200).json(users);
+      const { userID } = JwtService.verify(accessToken);
+      res.locals.loggedInUser = await UserService.getUser(userID);
```

Comment:
I didn't get the idea of storing the current user in `res.locals.loggedInUser`, and then later restoring the same into `req.user`? Can't we just store it directly into `req.user`?

Reply:
It's a 2 time check on if the user is actually using compromised jwt. If you notice, the user may or may not be stored in `res.locals.loggedInUser`. However it is guaranteed that we will find a user in `req.user`. That is why we have the additional check, in the `allowIfLoggedIn` method.

Also `res.locals` is historically being used for storing current req / res data. This is mostly for the view engines like `ejs`. They directly fetch the data from there and can also modify the data there. So we tend to save the same data else where. This is a case that will not be happening in our api project, as we are keeping the rest-api and the client's consuming the api separate still, I thought of following this practice. We can keep the `res.locals.loggedInUser` instead of `req.user` definitely.

Reply:
Okay!